std/typecons.d

```diff
@@ -1732,8 +1761,36 @@ succeeds, $(D this) becomes non-null.
     }
 
 /**
+    If this `Nullable` wraps a type that already has a null value
+    (such as a pointer), then assigning the null value to this
+    `Nullable` is no different than assigning any other value of
+    type `T`, and the resulting code will look very strange. It 
+    is strongly recommended that this be avoided by instead using
+    the version of `Nullable` that takes an additional `nullValue`
+    template argument.
+ */
+unittest
+{
+    //Passes
+    Nullable!(int*) npi;
+    assert(npi.isNull);
+    
+    //Passes?!
```

Comment:
The real question is, why would you put an already nullable value such as a pointer or class reference in a Nullable?
It might possibly happen in some generic code, but then assignment will likely also be generic, and special casing the behavior would counter-productive.
It seems your reasoning is flawed, b/c your trying to fix some behavior that looks weird but doesn't occur in practice and is a sound implementation.